HotelBooking/HotelBooking.DataAccessLayer/Database/HotelContext.cs

```diff
@@ -0,0 +1,114 @@
+﻿using Microsoft.EntityFrameworkCore;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace HotelBooking.DataAccessLayer.Database;
+
+public class HotelContext : DbContext
+{
+    public HotelContext()
+    {
+
+    }
+
+    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
+    {
+
+        optionsBuilder.UseSqlServer(@"Server=localhost,1433;Database=HotelBooking;Trusted_Connection=True;");
+
+    }
+
+    public DbSet<Dictionary<string, object>> Guests => Set<Dictionary<string, object>>("Guest");
+    public DbSet<Dictionary<string, object>> Reservations => Set<Dictionary<string, object>>("Reservation");
+    public DbSet<Dictionary<string, object>> Rooms => Set<Dictionary<string, object>>("Room");
+
+    protected override void OnModelCreating(ModelBuilder modelBuilder)
+    {
+        modelBuilder.Entity<Guest>(
+            g =>
```

Comment:
W tej sekcji (konfiguracja gościa pozbywamy się Aniu wszystkich HasColumnName, ponieważ nazwy jakie mają nasze właściwości podobają nam się i chcemy, żeby kolumny nazywały się tak samo. HasColumnName jest fajne jeśli chcielibyśmy nazywać kolumny inaczej niż nasze właściwości, my nie chcemy.
HaSColumnType też usuńmy, EF domyśli się :) Tym bardziej, że nie jestem przekonany do typu string na bazie. Z tego co wiem to w MSSQL nie ma takiego typu jak string. Migracja to puściła, ale nie wiem czy gdybyśmy chcieli już tworzyć bazę to czy nie pojawiłby się błąd. W sumie ciekawe, sprawdzę kiedyś. W każdym razie raczej oczekiwałbym dla wartości tekstowych pytu Varchar lub NVarchar i EF domyśli się tego, tak więc HasColumnType możemy usunąć.
Wychodzi na to, że usuneliśmy całą konfigurację dla Gościa :D. Co jednak mogibyśmy dodać ? Hmmm zastanówmy się..... Dobra już wiem:
na przykłam wszystkie nasze pola typu string zmapują się pewno tak jak mówiliśmy na Varchar (typ tekstowy w MSSQL), ale zmapują się na Varchar(max) co to max ? maksymalna dostępna liczba znaków. My nie potrzebujemy tylu znaków żeby zapisać mail lub telefon. To marnowanie pamięci żeby na 9 znakowy numer telefonu alogować tyle bajtów. Możemy więc użyć czegoś takiego: 
g.Property(g => g.PhoneNumber).HasMaxLength(9);
g.Property(g => g.Email).HasMaxLength(100);
Na mail dałem 100 znaków tak bezpiecznie, nie wiem w sumie jaki może być najdłuższy mail.
Inna rzecz która przychodzi mi do głowy to jest unikalność niektórych kolumn. Na przykład chcemy aby mogłoby być więcej niż 1 gość o takim samym mailu ? Wydaje mi sie, że nie. Mail powinien być wartością unikalną. Można więc dodać Unique na bazie:
g.HasIndex(g => g.Email).IsUnique();